hazelcast/src/main/java/com/hazelcast/config/YamlConfigBuilder.java

```diff
@@ -141,7 +162,139 @@ private void parseAndBuildConfig(Config config) throws Exception {
             throw new InvalidConfigurationException("No mapping with hazelcast key is found in the provided configuration");
         }
 
-        new YamlMemberDomConfigProcessor(true, config).buildConfig(W3cDomUtil.asW3cNode(imdgRoot));
+        Node w3cRootNode = asW3cNode(imdgRoot);
+        replaceVariables(w3cRootNode);
+        importDocuments(imdgRoot);
```

Comment:
No, `asW3cNode` actually wraps and returns a `Node` view of the provided YAML node. All mutations are applied to the wrapped YAML node(s).

Reply:
I see, that's always tricky when you deal with the mutable state. The preferred way would actually be to have the immutable objects and convert them instead of wrapping. But I guess it's too much work to change it that way. So you can leave it as it is.